### ISSUE TITLE
[Research] A reset button to restore filters to default state

### ISSUE THREAD

**Issue by donyunardi** [insightsengineering/teal.slice#298]
### Summary
As a user, I would like to restore the filters to their original default settings, just like when the app was first run.

### Acceptance Criteria
1. Conduct research to identify the necessary components and approaches for enabling the desired feature. 
2. Propose steps to complete the feature in small pieces. Breaking down the implementation into manageable steps can help ensure a systematic and organized approach

**Comment by lcd2yyz** [comment]
Propose to add a button with the turning black arrow icon at two levels
- beside the on/off toggle at the global filter panel level
- beside the collapse/expand accordion button at the dataset level

**Comment by donyunardi** [comment]
Just want to mention that the example app that @gogonzo mentioned in #165 (funny_module) already have reset button 😃 

![image](https://github.com/insightsengineering/teal.slice/assets/8597300/e4923f8d-5d21-433b-9087-ac401a857a32)

**Comment by chlebowa** [comment]
Please clarify:
+ @lcd2yyz OP refers to all filters or a single card, whereas you refer to one whole dataset
+ should the "undo" functionality (placeholder name) remember the most recent state or should it store history of all changes?

**Comment by lcd2yyz** [comment]
Can we try to come up with a design for as much flexibility as possible - meaning having a "reset" for each potential level of interaction
- For each filter card/variable
- For each module or module set (now that we can do module specific)
- At the global filter level (as if app have restarted)
On a second thought, resetting at each dataset-level probably does not make as much sense as resetting at module-level, so let's scratched that.

Once there is a rough design we can use to start discussion, we can go into the details of whether implementation is possible technically, and/or whether it make sense from UX perspective (too many buttons may not be a good thing either). Be as creative as you see fit.

> * Should we also consider an "undo" functionality (placeholder name) to remember the most recent state or store history of all changes? If so, should resetting clear history?

I wouldn't consider this to be a core requirement, but it will be very cool if possible! Do feel free to build that into the design as much as possible. My initial thought is "resetting" can either clear history, or be saved as another step in the history. Both can make sense depending on the actual usage, so I don't have a strong opinion either way at the moment.

**Comment by chlebowa** [comment]
All right. No dataset level then.

I must warn that module-level may well be impossible because - as @gogonzo pointed out - some filters are shared across modules. But perhaps we will come up with some mechanism.

**Comment by lcd2yyz** [comment]
Thanks for starting the design and thinking through the detail of the behaviors.

With the example you outlined, I agree "undo/back" action may not make sense at global level. We should definitely avoid changing filter settings unknowingly to the users. At global level, it should require more conscious interaction from the user to save some "states" perhaps (as oppose to save all states with undo). 
It would still be useful at individual filter card level, as users may be exploring different settings/thresholds. Will be cool if we can still retain this at filter-card level.

**Comment by chlebowa** [comment]
Yup, after some talks we agreed that it makes more sense to only be able to save snapshots globally, but perhaps track some history of filter cards in between global snapshots. That is what I am trying to figure out at the moment.
